src/client/Microsoft.Identity.Client/PlatformsCommon/DeviceAuthHelper.cs

```diff
@@ -76,8 +78,20 @@ public static bool CanOSPerformPKeyAuth()
         {
             //PKeyAuth can only be performed on operating systems with a major OS version of 6. 
             //This corresponds to windows 7, 8, 8.1 and their server equivilents.
-#if DESKTOP
-            if (Environment.OSVersion.Version.Major == 6)
```

Comment:
Sense Environment.OSVersion is not accurate, I used slightly different methods to properly detect the OS version. These are slightly different between .net fwk and .net core but produce the same results in testing

Reply:
@pmaytak We used to advertise that we are PKey Auth capable on every call in the token client but now we are looking to only do this on older versions of windows. (win 8 and below).

Thus I am adding these checks here to make sure we are only adding this header on supported platforms. Do you think these checks will have any impact on performance?

Reply:
@pmaytak We used to advertise that we are PKey Auth capable on every call in the token client but now we are looking to only do this on older versions of windows. (win 8 and below).

Thus I am adding these checks here to make sure we are only adding this header on supported platforms. Do you think these checks will have any impact on performance?

Reply:
@trwalke Seems like `CanOSPerformPKeyAuth` ends up being called for each request, right? Also, the OS version shouldn't change. So I would probably call this method once (at app startup or with a first request) and cache the OS version in a variable somewhere in the app.

Also according to [this](https://stackoverflow.com/questions/1097363/what-are-the-performance-implications-of-accessing-a-windows-registry-value) and [this](https://devblogs.microsoft.com/oldnewthing/20060222-11/?p=32193), registry access should be fast and the recent values are cached, but there're still extra those cycles spent, compared to if we cache the value in the app.

Reply:
Please add the link where you copied the registry solution to the code, it's helpful to have a reference